### PR TITLE
update deps and tidy up

### PR DESCRIPTION
**Pull request by zarsky-broad**

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -32,11 +32,14 @@ server:
 
 spring:
   application.name: workspace
-  resources:
-    cache.cachecontrol:
-      max-age: 0
-      must-revalidate: true
-    static-locations: classpath:/api/
+  web:
+    resources:
```
Comment (zarsky-broad) [inline src/main/resources/application.yml:36]: old path is deprecated in new spring boot

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -32,11 +32,14 @@ server:
...
+      cache:
+        cachecontrol:
+          max-age: 0
+          must-revalidate: true
+        use-last-modified: false
```
Comment (zarsky-broad) [inline src/main/resources/application.yml:41]: newly available

File: src/test/java/bio/terra/workspace/common/BaseTest.java [hunk 3]
```diff
@@ -7,7 +7,7 @@
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
-@ActiveProfiles("test")
+@ActiveProfiles({"test", "human-readable-logging"})
```
Comment (zarsky-broad) [inline src/test/java/bio/terra/workspace/common/BaseTest.java:10]: old include in application-test.yml wasn't working reliably

**Review by zloery — APPROVED** [review]
LGTM, though I haven't tested any of the swagger cache controls.

**Comment by zarsky-broad** [comment]
FWIW, this shouldn't really change anything about how the caching is happening; it might be possible to remove some of our other cache-busting code with this addition, but for the moment I'm just adding it.

File: build.gradle [hunk 4]
```diff
@@ -2,13 +2,13 @@ plugins {
...
-	id "com.diffplug.spotless" version "5.7.0"
+	id "com.diffplug.spotless" version "5.8.2"
 	id "com.github.ben-manes.versions" version "0.36.0"
-	id "com.google.cloud.tools.jib" version "2.6.0"
+	id "com.google.cloud.tools.jib" version "2.7.0"
 	id "com.palantir.git-version" version "0.12.3"
 	id "io.spring.dependency-management" version "1.0.10.RELEASE"
 	id "org.hidetake.swagger.generator" version "2.18.2"
-	id "org.springframework.boot" version "2.3.5.RELEASE"
+	id "org.springframework.boot" version "2.4.0"
```
Comment (davidangb) [inline build.gradle:11]: have you reviewed the 2.4 release notes? There's some stuff in there around config loading, k8s config integration, and docker image publishing that could be interesting and/or important.

**Review by davidangb — COMMENTED** [review]
The combination of your comment that "old include in application-test.yml wasn't working reliably" and Spring's comment that "If, however, you’ve have a more complex setup (with profile-specific properties, or profile activation properties) you may need to make some changes if you want to use the new features." (https://spring.io/blog/2020/11/12/spring-boot-2-4-0-available-now) makes me worry. Have you read through all those notes?

**Comment by zarsky-broad** [comment]
@davidangb Ah, I read the https://github.com/spring-projects/spring-boot/releases/tag/v2.4.0 notes, but not the posts, good catch!

**Comment by zarsky-broad** [comment]
@davidangb Read through those notes (and the migration guide, and...), and I'm sticking with the changes in this PR.

The alternative would be to define a profile group, i.e.
```yaml
spring.profiles.group.test: ["human-readable-logging"]
```
but that would have to be defined in the base `application.yaml`, and I don't want to put test logic there.

File: build.gradle (hunk 4 above)
Comment (zarsky-broad) [reply build.gradle:11]: I don't think any of the things other than the config include change is relevant.

**Review by davidangb — APPROVED** [review]
thank you for the due diligence on this one!

**Pull request merged**